src/System Application/App/Cryptography Management/src/RSACryptoServiceProviderImpl.Codeunit.al

```diff
@@ -32,6 +32,17 @@ codeunit 1446 "RSACryptoServiceProvider Impl." implements "Signature Algorithm v
         DotNetAsymmetricAlgorithm := DotNetRSACryptoServiceProvider;
     end;
 
+    [NonDebuggable]
+    procedure CreateRSAKeyPair(var PublicKeyInXML: Text; var PrivateKeyInXML: SecretText)
+    var
+        DotnetRSA: DotNet RSA;
+    begin
+        RSACryptoServiceProvider();
+        DotnetRSA := DotNetRSACryptoServiceProvider.Create();
+        PublicKeyInXML := DotnetRSA.ToXmlString(false);
+        PrivateKeyInXML := DotnetRSA.ToXmlString(true);
```

Comment:
From what I can read, this is not just the private key but both the public and private key: https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.rsa.toxmlstring?view=net-8.0

Reply:
depends on what you give as the parameter. See includePrivateParameters in the link you provided.

Reply:
Yeah, that's the one I'm referencing:
<img width="253" alt="image" src="https://github.com/microsoft/BCApps/assets/10121161/5b098d15-6bb6-4e42-9033-dc9f7ed1a355">
Not a big deal though, just that it contains both :)

Reply:
The PrivateKeyXmlStringSecret used in the tests is containing exactly that, both pub and private. That is how their implementation works, also when you call FromXMLString().

Reply:
This is a non-issue. Resolving.